Branches/stable 1.12

machadovilaca opened the pull request.

mdounin commented:
Please note that this repository is unofficial and pull requests have no chances to be merged. Proper way to submit changes to nginx is via the nginx development mailing list, see http://nginx.org/en/docs/contributing_changes.html.

mdounin committed:
Stable branch.

mdounin committed:
release-1.12.0 tag

mdounin committed:
nginx-1.12.0-RELEASE

mdounin committed:
Updated OpenSSL used for win32 builds.

mdounin committed:
Range filter: protect from total size overflows.

The overflow can be used to circumvent the restriction on total size of
ranges introduced in c2a91088b0c0 (1.1.2).  Additionally, overflow
allows producing ranges with negative start (such ranges can be created
by using a suffix, "bytes=-100"; normally this results in 200 due to
the total size check).  These can result in the following errors in logs:

[crit] ... pread() ... failed (22: Invalid argument)
[alert] ... sendfile() failed (22: Invalid argument)

When using cache, it can be also used to reveal cache file header.
It is believed that there are no other negative effects, at least with
standard nginx modules.

In theory, this can also result in memory disclosure and/or segmentation
faults if multiple ranges are allowed, and the response is returned in a
single in-memory buffer.  This never happens with standard nginx modules
though, as well as known 3rd party modules.

Fix is to properly protect from possible overflow when incrementing size.

mdounin committed:
Version bump.

mdounin committed:
nginx-1.12.1-RELEASE

mdounin committed:
release-1.12.1 tag
